### ISSUE TITLE
"Nested[Y@ListTransform]" is not a subtype of "Nested[Y@ListTransform]"

### ISSUE THREAD

**Issue by randolf-scholz** [microsoft/pyright#8602]
Code sample in [pyright playground](https://pyright-play.net/?strict=true&code=MYGwhgzhAEAqBOYB2EBmB7eBbA2gDQBpoBNAXQC4AoaG6AEwFNVoB9FgSyQDcH4AXNgAoIDEKgCU0ALQA%2BaACIEyNJlzEieUvPLQAdPsqU%2BATwAODaADkGEPgzo5YpaAF440AD7QQ7WznnWtvaOWqSGoJAwADK%2BfEooGNj4RGSC8SpJgXYOmkRZwWSk4lS09EysHNy8AizCohLScvIxtumJahpaOvq6lEA)

```python
class Transform[X, Y]:
    def __invert__(self) -> "Transform[Y, X]": ...

type Nested[T] = T | list["Nested[T]"]

class ListTransform[X, Y](Transform[Nested[X], Nested[Y]]):
    def __invert__(self) -> "ListTransform[Y, X]": ...
```

```
Method "__invert__" overrides class "Transform" in an incompatible manner
  Return type mismatch: base method returns type "Transform[Nested[Y@ListTransform], Nested[X@ListTransform]]", override returns type "ListTransform[Y@ListTransform, X@ListTransform]"
    "ListTransform[Y@ListTransform, X@ListTransform]" is incompatible with "Transform[Nested[Y@ListTransform], Nested[X@ListTransform]]"
      Type parameter "X@Transform" is covariant, but "Nested[Y@ListTransform]" is not a subtype of "Nested[Y@ListTransform]"
      Type parameter "Y@Transform" is covariant, but "Nested[X@ListTransform]" is not a subtype of "Nested[X@ListTransform]"  (reportIncompatibleMethodOverride)
```

Maybe noteworthy: for `pyright<=1.1.372` the message is different:

```
Type parameter "X@Transform" is covariant, but "Y@ListTransform | list[Nested]" is not a subtype of "Nested[Y@ListTransform]"
Type parameter "Y@Transform" is covariant, but "X@ListTransform | list[Nested]" is not a subtype of "Nested[X@ListTransform]"
```

**Comment by erictraut** [comment]
This was a long-standing issue that just so happens to be fixed by a change I made yesterday. This fix will be included in this week's release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.374.
